### PR TITLE
Cross platform clients packaged in RPM form

### PR DESCRIPTION
**Pull request by sdodson**
This adds golang cross compiled binaries to the RPM build process.

### CODE REVIEW

**Comment by sdodson** [comment]
This may be of little value to origin but this is how Red Hat's release and configuration management team has requested that we deliver windows and mac clients to them for inclusion in Red Hat Customer Portal. These binaries could also be exposed via the console or something too for download.

File: openshift.spec [hunk 1]
```diff
@@ -94,12 +105,19 @@ pushd _thirdpartyhacks
...
-    go build -ldflags "%{ldflags}" %{import_path}/cmd/${cmd}
+  export GOOS=${OS}
+  for cmd in openshift
+  do
+    if [ $GOOS == 'windows' ]; then
+      export GOARCH='386'
+    else
+      export GOARCH='amd64'
+    fi
+      go install -ldflags "%{ldflags}" %{import_path}/cmd/${cmd}
```
Comment (brenton) [inline openshift.spec]: I think the indentation is off a little here.

**Comment by sdodson** [comment]
@brenton look better now? What do you think about eliminating the loop and conditionals and just staticly listing the builds?

**Comment by brenton** [comment]
Yeah.

I'm OK with the loop personally.  I'm sure we'll have to revisit that code in the future as things get more complex though.  Want me to submit this for merging?

**Comment by sdodson** [comment]
Yeah, if you're ok with it lets merge.

**Comment by brenton** [comment]
[merge]

**Comment by sdodson** [comment]
@brenton Seems like a flakey test given the test passed but the merge failed. re-merge?

**Comment by brenton** [comment]
[merge]

**Pull request merged**